protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/FileBuilder.kt

```diff
@@ -180,7 +190,9 @@ internal open class FileBuilder(val namer: Namer = Namer.Standard, val supportMa
                 localTypeName = fieldDesc.typeName,
                 repeated = fieldDesc.label == FieldDescriptorProto.Label.REPEATED,
                 jsonName = fieldDesc.jsonName,
-                optional = !alwaysRequired && fieldDesc.label == FieldDescriptorProto.Label.OPTIONAL,
+                optional = !alwaysRequired &&
+                        ((fieldDesc.label == FieldDescriptorProto.Label.OPTIONAL && ctx.fileDesc.syntax.let{it == "proto2" || it == null}) ||
```

Comment:
I'd recommend defining an extension property on `FieldDescriptorProto` to make the syntax check easier to understand. Something like:

```kt
private val FieldDescriptorProto.usesProto2Syntax: Boolean get() = syntax == null || syntax == "proto2"
```

Then the code here (and the other spot in this file where you check the syntax) would read as `ctx.fileDesc.usesProto2Syntax`.